### PR TITLE
remove unnecessary calculation in Vector3.project

### PR DESCRIPTION
**Pull request by zhaoy-dev**
remove unnecessary calculation in Vector3.project

### CODE REVIEW

**Comment by yaoyao-cn** [comment]
i test with my chrome, it's about 20% faster

File: src/math/Vector3.js [hunk 1]
```diff
@@ -312,27 +312,19 @@ Object.assign( Vector3.prototype, {
 
 	},
 
-	project: function () {
+	project: function ( camera ) {
 
-		var matrix = new Matrix4();
-
-		return function project( camera ) {
+		return this.applyMatrix4( camera.matrixWorldInverse ).applyMatrix4( camera.projectionMatrix );
```
Comment (Mugen87) [inline src/math/Vector3.js]: Are you sure using `camera.matrixWorldInverse` is safe? I wonder if there are scenarios where this matrix might not be up to date (or in sync with `camera.matrixWorld`)...

Comment (yaoyao-cn) [reply src/math/Vector3.js]: yes, i am sure. camera.matrixWorldInverse was updated by the camera itself automatically in Camera.updateMatrixWorld. and it should be sync with camera.matrixWorld as the name implies.

**Review by Mugen87 — APPROVED** [review]

File: src/math/Vector3.js (hunk 1 above)
Comment (mrdoob) [reply src/math/Vector3.js]: Yep! https://github.com/mrdoob/three.js/blob/dev/src/cameras/Camera.js#L60

**Comment by mrdoob** [comment]
Thanks!

**Comment by mrdoob** [comment]
Now I feel that we need `camera.projectionMatrixInverse` to make this code prettier 😅

**Pull request merged**